plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHABase.java

```diff
@@ -33,7 +35,7 @@ public class KVMHABase {
     private long _timeout = 60000; /* 1 minutes */
     protected static String s_heartBeatPath;
     protected long _heartBeatUpdateTimeout = 60000;
-    protected long _heartBeatUpdateFreq = 60000;
+    protected long _heartBeatUpdateFreq = AgentPropertiesFileHandler.getPropertyValue(AgentProperties.KVM_HEARTBEAT_UPDATE_FREQUENCY);
```

Comment:
(1) The focus of this PR is only the `heartBeatUpdateFreq`. What other variables would you consider necessary to externalize?
(2) Yes, this will be fixed.

Reply:
> (1) The focus of this PR is only the `heartBeatUpdateFreq`. What other variables would you consider necessary to externalize? (2) Yes, this will be fixed.

@stephankruggg 
good. 
can you discuss with your team if other fixed values need be configurable as well ?
if you want only this configurable, it is ok.
if you want more to be configurable, you can make all the changes in this PR.

Reply:
@weizhouapache, speaking as a member of the SC Clouds team, every PR opened by someone from SC Clouds already attends to the use case the company faced. In this case, externalizing the property `kvm.heartbeat.update.frequency` already attends the SC Clouds use case. Therefore, if there are improvements to be done or other cases to be considered within the proposal's boundaries, the community should point them out; otherwise, if the community thinks it is ok as it is, then there is nothing else to address.

Now, speaking as a member of the community, once a PR is opened, it does not matter from which company the contributor who opened the PR is; it is a contribution from someone to the community and should be treated as so. The bond is not the company with the community, is the contributor with the community; the contributor's team is the community. Then, when @stephankruggg asked you `what other variables would you consider necessary to externalize` he was already discussing with the team (the community) if other variables should be externalized. Also, we have to remember that not all contributions are based on a company's motivation; contributors want the best for the project and may contribute to the community's evolution independently of a company saying them to do so.

Reply:
@GutoVeronezi 
I think I have clearly expressed my opinion. I have no objection if you want only `kvm.heartbeat.update.frequency` to be configurable, it is ok to me. I also noticed that there are some other related properties can be configurable as well. Do you need me to point them out ? I believe you have noticed them too.

This might happen in the future: the heartbeat update timeout (60seconds) or a setting else is not good, then externalize it to agent.properties. For your company, you need to develop, review, test, release and deploy it again. The community needs to review and test again. It is good if you have discussed it. Hope it will not come true. 

You can of course submit your PRs as individual contributors. The fact we must face is, you/we are also part of a company, most of our changes are based on the company's requirements. The rule of merging PR is, PR can only be merged with 2 approvals (including 1 verification unless it is not required) and smoke tests pass. If you look at the PRs merged recently, most have been verified by @DaanHoogland. However, the community does not have resources to manually test every PR ( I know some PRs have been merged without manual verification, which is not good). I personally encourage the companies which have good resources to test their PRs and share the results, so both the company and the community can benifit from their contribution (including not only creating PRs, but also discussing/reviewing/testing PRs).

Back to the topic, @stephankruggg can you explain a bit why you made this setting configurable, but not for other settings ?

Reply:
@GutoVeronezi 
you are probably right. I should mention individual contributors who are interested and benefit more from the changes, for example you, João, Rodrigo, Gabriel, Stephan, SadiJr. Hope you are ok with it.

Again, if contributors want to merge a PR as quick as possible, please ask others to test it. Most PRs should not be merged without manual verification (two LGTM/approvals are not good enough).

Reply:
before this becomes a an even more big discussion: let's stick to the question of whetther we keep it to this setting and go through the trouble of testing new PRs when we want other setting, or we add as much settings as we can now?
I see the rest of the discussion as useful, no doubt but it kind of transcends this PR.

Reply:
The intention of this PR is to allow the increase (or the decrease) of the frequency of the KVM HA heartbeat checks, which is currently not possible. For example, if one wants to guarantee that a critical host is down for the least amount of time possible, one can increase the frequency of the heartbeat with the changes proposed.

Even though not necessarily related to the same use case, the externalization of other properties would achieve the same general goal of increasing the flexibility of the system. Since everyone benefits from this, and they are all related to the same topic (KVM HA), I am open to work on the externalization of them in this PR.

Reply:
thanks @stephankruggg , I'll await your ping to work on merging this.